pkg/daemon/daemon.go

```diff
@@ -413,16 +413,26 @@ func (dn *Daemon) nodeStateSyncHandler() error {
 		log.Log.Error(err, "nodeStateSyncHandler(): Failed to fetch node state", "name", vars.NodeName)
 		return err
 	}
-	latest := latestState.GetGeneration()
-	log.Log.V(0).Info("nodeStateSyncHandler(): new generation", "generation", latest)
+
+	log.Log.V(0).Info("nodeStateSyncHandler(): new generation", "generation", latestState.GetGeneration())
 
 	if vars.ClusterType == consts.ClusterTypeOpenshift && !dn.platformHelpers.IsHypershift() {
 		if err = dn.getNodeMachinePool(); err != nil {
 			return err
 		}
 	}
 
-	if dn.nodeState.GetGeneration() == latest {
+	// load plugins if it has not loaded
+	if len(dn.enabledPlugins) == 0 {
+		dn.enabledPlugins, err = enablePlugins(latestState, dn.HostHelpers)
+		if err != nil {
+			log.Log.Error(err, "nodeStateSyncHandler(): failed to enable vendor plugins")
+			return err
+		}
+	}
+
+	// TODO: include this logic specific of systemd into shouldSkipReconciliation method. Previous workflow hasn't been modified. This part is executed first.
```

Comment:
any reason not to move it ?

OR if you prefer to keep this here, remove the TODO.

I prefer not to have this comment in code.

Reply:
I will remove the TODO. The logic around systemd is easier to understand with that part outside I think